bin/mongodb-schema

```diff
@@ -133,93 +128,101 @@ function getTable(schema) {
   return table;
 }
 
-var bar = new ProgressBar('analyzing [:bar] :percent :etas ', {
+const bar = new ProgressBar('analyzing [:bar] :percent :etas ', {
   total: argv.number * argv.repeat,
   width: 60,
   complete: '=',
   incomplete: ' ',
   clear: true
 });
 
-mongodb.connect(uri, {useUnifiedTopology: true}, function(err, conn) {
-  if (err) {
-    console.error('Failed to connect to MongoDB: ', err);
-    process.exit(1);
+const client = new MongoClient(uri);
+
+(async function main() {
+  try {
+    await client.connect();
+  } catch (err) {
+    if (err) {
+      console.error('Failed to connect to MongoDB: ', err);
+      process.exit(1);
+    }
   }
 
-  var ns = toNS(argv._[1]);
-  var db = conn.db(ns.database);
-  var ts;
+  const ns = toNS(argv._[1]);
+  const db = client.db(ns.database);
+  let ts;
 
-  var options = {
+  const options = {
     size: sampleSize,
     query: {},
     promoteValues: argv.promote
   };
 
-  var schema;
-  var schemaOptions = {
+  let schema;
+  const schemaOptions = {
     storeValues: argv.values,
     semanticTypes: argv.semanticTypes
   };
 
-  async.timesSeries(argv.repeat, function(arr, cb) {
-    var source = argv.sampling ?
-      sample(db, ns.collection, options) :
-      db.collection(ns.collection).find(options.query, {}, {
-        promoteValues: options.promoteValues
-      }).limit(options.size);
-
-    source
-      .once('data', function() {
-        ts = new Date();
-      })
-      .pipe(schemaStream(schemaOptions))
-      .on('progress', function() {
-        bar.tick();
-      })
-      .on('data', function(data) {
-        schema = data;
-      })
-      .on('end', function() {
-        var dur = new Date() - ts;
-        cb(null, dur);
+  try {
+    for (let i = 0; i < argv.repeat; i++) {
+      await new Promise((resolve, reject) => {
+        const source = argv.sampling
+          ? sample(db, ns.collection, options)
```

Comment:
To get this `sample` working we'll have to update the `mongodb-collection-sample` package to work with a newer MongoDB version:
https://github.com/mongodb-js/collection-sample
In particular the newer cursors don't supply `.pipe(...` and instead we can use `.stream(...`
Before spending the time to do that I want to get a sense of how widely used this command line interface is used; if it isn't being used we might be able to drop support of this interface to the `mongodb-schema` package.

Reply:
done